### DISCUSSION TITLE
How are we building the image that is pushed in quay

### DISCUSSION THREAD

**Discussion by crombus** [keycloak/keycloak#23167] [Q&A]
I need to update the keycloak image in mcr of microsoft.

I just need simple image no operator or quarkus image, how we are building

**Comment by vmuzikar** [comment]
Please see https://github.com/keycloak/keycloak/blob/main/quarkus/container/README.md

**Comment by crombus** [comment]
Thanks for the reply .

But what is ` KEYCLOAK_DIST=http://<HOST>:<PORT>/keycloak-<VERSION>.tar.gz . -t <YOUR_TAG>`

how to get the tar file and what should be fixed in the variable.

**Reply by vmuzikar** [reply]
That's for pointing the Docker build to an existing Keycloak dist hosted somewhere.

You can also point it to a local archive, something like (provided you copied the tar to the same location as is the Dockerfile):
```
docker build --build-arg KEYCLOAK_DIST=$(ls keycloak-*.tar.gz) . -t keycloak:test
```

**Reply by crombus** [reply]
How can I build the keycloak Dist any steps for that.

https://github.com/keycloak/keycloak/tree/22.0.3/operator#building

This command take very long to build and I have issues with it https://github.com/keycloak/keycloak/discussions/23257#discussion-5627613

**Reply by vmuzikar** [reply]
To build the server dist ZIP, you can follow https://github.com/keycloak/keycloak/blob/main/docs/building.md

**Comment by crombus** [comment]
I build the jar file using this command : ` mvn clean package -Dquarkus.container-image.build=true -DskipTests`

it was success :
[INFO] --- maven-resources-plugin:3.2.0:resources (default-resources) @ keycloak-examples-ldap ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ keycloak-examples-ldap ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:3.2.0:testResources (default-testResources) @ keycloak-examples-ldap ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ keycloak-examples-ldap ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ keycloak-examples-ldap ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-war-plugin:3.2.3:war (default-war) @ keycloak-examples-ldap ---
[INFO] Packaging webapp
[INFO] Assembling webapp [keycloak-examples-ldap] in [/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/ldap-portal]
[INFO] Processing war project
[INFO] Copying webapp resources [/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/src/main/webapp]
[INFO] Webapp assembled in [2 msecs]
[INFO] Building war: /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/ldap-portal.war
[INFO] 
[INFO] --- maven-source-plugin:3.1.0:jar-no-fork (attach-sources) @ keycloak-examples-ldap ---
**[INFO] Building jar: /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/ldap-portal-sources.jar**




while building the image using command : docker build --build-arg KEYCLOAK_DIST="/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/ldap-portal-sources.jar" . -t keycloak:test


I got got the error

ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount3815879342/Dockerfile: no such file or directory

**Reply by vmuzikar** [reply]
You need to run the `docker build` command inside `quarkus/container`.

Also
```
KEYCLOAK_DIST="/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/examples/ldap/target/ldap-portal-sources.jar"
```
does not seem correct. It needs to point to a tar with the distribution.

**Reply by crombus** [reply]
these commands should build the tar file right

  git clean -xdf
  mvn clean package -Dquarkus.container-image.build=true -DskipTests


after this docker build is needed.

**Reply by crombus** [reply]
I am in the quarkus dir but I get the error
` ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2249683113/Dockerfile: no such file or directory`

**Reply by vmuzikar** [reply]
Would you be able to share specific steps to reproduce?

**Reply by crombus** [reply]
Steps:
                    cd quarkus
                    ../mvnw clean install -DskipTests
                    docker images
                    ls -la /mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/quarkus/dist/target/
                    pwd
                    cd container
                    docker build --build-arg KEYCLOAK_DIST="/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/quarkus/dist/target/keycloak-22.0.2.tar.gz" . -t keycloak:test
                    docker image ls
                    
                    
Error:
8 ERROR: failed to calculate checksum of ref a2836209-1a7a-4178-8ee5-d44ff1e7f696::xi6w2q3dz2yd4sy6u96bhpjrd: "/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/quarkus/dist/target/keycloak-22.0.2.tar.gz": not found


Files in location dist/target
drwxr-xr-x 7 cloudtest cloudtest      4096 Sep 28 05:33 .
drwxr-xr-x 4 cloudtest cloudtest      4096 Sep 28 05:33 ..
drwxr-xr-x 2 cloudtest cloudtest      4096 Sep 28 05:33 archive-tmp
drwxr-xr-x 2 cloudtest cloudtest      4096 Sep 28 05:33 dependency-maven-plugin-markers
-rw-r--r-- 1 cloudtest cloudtest 164637881 Sep 28 05:33 keycloak-22.0.2.tar.gz
-rw-r--r-- 1 cloudtest cloudtest 164713037 Sep 28 05:33 keycloak-22.0.2.zip
drwxr-xr-x 3 cloudtest cloudtest      4096 Sep 28 05:33 keycloak-client-tools
drwxr-xr-x 5 cloudtest cloudtest      4096 Sep 28 05:33 keycloak-quarkus-server
drwxr-xr-x 4 cloudtest cloudtest      4096 Sep 28 05:33 keycloak-quarkus-server-app
/mnt/vss/_work/1/go/src/github.com/keycloak/keycloak/quarkus

**Reply by crombus** [reply]
Output of cmd ../mvnw clean install -DskipTests

[INFO] Reactor Summary for Keycloak Quarkus Parent 22.0.2:
[INFO] 
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [ 25.220 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [ 17.339 s]
[INFO] Keycloak Quarkus Server Extension .................. SUCCESS [ 36.908 s]
[INFO] Keycloak Quarkus Server Deployment ................. SUCCESS [  2.006 s]
[INFO] Keycloak Quarkus Server App ........................ SUCCESS [  8.539 s]
[INFO] Keycloak Quarkus Server Distribution ............... SUCCESS [ 21.755 s]
[INFO] Keycloak Quarkus Test Parent ....................... SUCCESS [  0.006 s]
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SUCCESS [  1.683 s]
[INFO] Keycloak Quarkus Server Integration tests .......... SUCCESS [  2.818 s]

**Reply by vmuzikar** [reply]
You have to [copy the distribution](https://github.com/keycloak/keycloak/blob/1f44474aac75ffc5fa5f0a3a476a1c9c330f2f5b/quarkus/container/README.md?plain=1#L12) to your Docker build context root.
